uber_sante/controllers/booking_controller.py

```diff
@@ -29,25 +32,28 @@ def book():
     # TODO: connect the call to the scheduler class to try reserving the availability (line 51)
     # TODO: connect the call to the booking_service to create the booking in the Booking table (line 54)
 
-        appointment_id = request.args.get('appointment_id')
+        availability_id = request.args.get('availability_id')
         patient_id = request.args.get('patient_id')
 
-        if appointment_id is None:
-            return jsonify('No appointment specified'), 400
+        if availability_id is None:
+            return js.create_json(data=None, message="No appointment specified", return_code=js.ResponseReturnCode.CODE_400)
         if patient_id is None:
-            return jsonify('No patient specified'), 400
+            return js.create_json(data=None, message="No patient specified", return_code=js.ResponseReturnCode.CODE_400)
+
 
-        patient = get_from_cache(patient_id) # get the patient from cache
-        appointment = patient.cart[appointment_id] # somewhat of a placeholder for now, waiting for cart to be implemented
+        # TODO: Remove Dummy appointment below and use the implementation below
+        patient = get_from_cache(patient_id)  # get the patient from cache
+        appointment = patient.cart.get_appointment(availability_id)
+
+        result = Scheduler.get_instance().reserve_appointment(appointment)
```

Comment:
OK, can you please change the [interaction diagram here ](https://www.draw.io/?state=%7B%22ids%22:%5B%221eoU0QSbH-nXKBlBUiZ7wLbwOqBMm_Qsd%22%5D,%22action%22:%22open%22,%22userId%22:%22111941182143072512528%22%7D#G1eoU0QSbH-nXKBlBUiZ7wLbwOqBMm_Qsd)to follow this new way of reserving the appointment.

Reply:
@Lussier115 the interaction diagram parameter is still incorrect for reserve_appointment